src/components/Organisms/TodoCard/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+import AddTaskButton from "../../Atoms/AddTaskButton/index";
+import Task from "../../Molecules/Task/index";
+import COLOR from "../../../variables/color";
+
+const TodoCard = () => {
+  const [todoTask, setTodoTask] = useState([]);
+  const [taskAddOREdit, setAddOREdit] = useState(false);
+
+  let todoTaskList = [];
+  let todoStateTask = [];
+
+  const handleTaskComplete = (id) => {
+    let completedTask = todoTask.map((task) => {
+      if (task.id === id) {
+        task.state = "DONE";
+      }
+      return task;
+    });
+    setTodoTask(completedTask);
+  };
+
+  const handleEditComplete = (id, taskName) => {
+    let edittedTask = [];
+    if (taskName === "") {
+      edittedTask = todoTask.filter((task) => {
+        return task.id !== id;
+      });
+    } else {
+      edittedTask = todoTask.map((task) => {
+        if (task.id === id) {
+          task.name = taskName;
+        }
+        return task;
+      });
+    }
+    setTodoTask(edittedTask);
+    setAddOREdit(false);
+  };
+
+  // task.stateがTODOのものだけ抽出する
+  todoStateTask = todoTask.filter((task) => {
+    return task.state === "TODO";
+  });

```

Comment:
```suggestion
const  todoStateTask = todoTask.filter((task) => {
    return task.state === "TODO";
  });
```
11行目の変数宣言と分ける意味がないのでここにまとめて書きましょう。

プログラミング中に手続型と関数型という考え方があり、reactは関数型の考え方に基づいたライブラリだと言われています。
関数型の考えに則れば、変数への再代入は避けたほうがいいものとなります。

以下は少し高度な記事ですが、参考になりそうです。
https://zenn.dev/tockri/books/dcaf6c55e64448/viewer/react